s3fs/core.py

```diff
@@ -2429,6 +2436,26 @@ def _abort_mpu(self):
             self.mpu = None
 
 
+# Define a new class to represent the file-like object for writing
+class S3AsyncStreamWriter:
+    def __init__(self, s3_fs, bucket, key):
+        self.s3_fs = s3_fs
+        self.bucket = bucket
+        self.key = key
+        self.closed = False
+        self.loc = 0
+
+    async def write(self, data):
+        # Write data directly to S3 object
+        await self.s3_fs._call_s3(
```

Comment:
This call will overwrite the file on each call to .write(). Ideally, we'd want to be able to push to an open HTTP stream, but that doesn't seem possible. However, s3 allows for multi-part uploads, with the condition that each partc is >=5MB. So we also would need buffering of bytes until there are enough.